src/app/shared/components/comment-form/comment-form.component.ts

```diff
@@ -48,44 +49,61 @@ export class CommentFormComponent implements OnInit {
   ) {}
 
   ngOnInit(): void {
+    this.updateFormType();
     if (this.data.comment != null) {
-      this.title = 'Uredi komentar';
-      this.commentForm.patchValue({ content: this.data.comment.content });
-
-      if (this.data.comment.type === 'warning') {
-        this.addExposedUntilField();
-        this.commentForm.patchValue({
-          exposedUntil: this.data.comment.exposedUntil,
-        });
-      }
+      this.commentForm.patchValue({
+        content: this.data.comment.content,
+        type: this.data.comment.type,
+      });
     }
+  }
 
-    switch (this.data.type) {
+  onTypeChange() {
+    this.updateFormType();
+  }
+
+  updateFormType() {
+    let prefix = this.data.comment != null ? 'Uredi' : 'Dodaj';
+
+    switch (this.type) {
+      case 'comment':
+        this.title = `${prefix} komentar`;
+        this.removeExposedUntilField();
+        break;
       case 'warning':
-        this.title = 'Dodaj opozorilo';
+        this.title = `${prefix} opozorilo`;
         this.addExposedUntilField();
+        if (this.data.comment != null) {
+          this.commentForm.patchValue({
+            exposedUntil: this.data.comment.exposedUntil,
+          });
+        }
         break;
       case 'condition':
-        this.title = 'Dodaj informacijo o razmerah';
-        break;
-      case 'comment':
-        this.title = 'Dodaj komentar';
+        this.title = `${prefix} opis razmer`;
+        this.removeExposedUntilField();
         break;
       case 'description':
         // TODO: it seems that only peaks and routes can have comments of type description. Might want to include entity type in the title and generalize this. Should discuss.
-        this.title = 'Dodaj opis';
+        this.title = `${prefix} opis smeri`;
+        this.removeExposedUntilField();
         break;
-      default:
-        this.title = '';
     }
   }
 
   addExposedUntilField() {
-    this.commentForm.addControl('exposedUntil', new FormControl());
+    this.commentForm.addControl(
+      'exposedUntil',
+      new FormControl(null, [Validators.required])
+    );
     this.maxDate = new Date();
     this.maxDate.setMonth(this.maxDate.getMonth() + 1); // let user choose max 1 month validity of warning exposure
   }
 
+  removeExposedUntilField() {
+    this.commentForm.removeControl('exposedUntil');
+  }
```

Comment:
This form control now also needs to be removed because it is a required field of this form. If not removed, it would prevent the user from submitting in case they first selected a `warning` type and then some other type.

Reply:
Also resolved with https://github.com/plezanje-net/web/pull/402/commits/8e575ae87fdfb77659d1fc737c0a720acaaee670

But we should anyway hide the control (and empty it) in case that we are not posting a warning.

I wonder if hiding it instead of removing it would make for less of flickering when changing comment type in the popup.